compiler/codegen/FrontEnd.hpp

```diff
@@ -125,6 +125,7 @@ class TR_FrontEnd : public TR_Uncopyable
    // Method
    // --------------------------------------------------------------------------
 
+   virtual bool handleExceptionThrownFromILGen();   // This API is to be used in a catch block
```

Comment:
If we plan to rethrow in the catch block instead of in the frontend, we can not try catch the exception in the frontend, thus we will not be able get the exception details. If we want to treat all the exceptions the same, then yes, we can make the frontend return an enum. @andrewcraik What do you think?